File: pet-clinic-scala/src/main/scala/com/github/rskupnik/repositories/CustomerRepository.scala
package com.github.rskupnik.repositories

import com.github.rskupnik.model.Customer
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

// Need to specify java.lang.Long instead of Scala's Long for some reason

@Repository
trait CustomerRepository extends CrudRepository[Customer, java.lang.Long] {
  def findByLastName(lastName: String): List[Customer]
}

File: pet-clinic-java/src/main/java/com/github/rskupnik/petclinic/repositories/CustomerRepository.java
package com.github.rskupnik.petclinic.repositories;

import com.github.rskupnik.petclinic.model.Customer;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface CustomerRepository extends CrudRepository<Customer, Long> {
    List<Customer> findByLastName(String lastName);
}
